mail_activity_partner/models/mail_activity.py

```diff
@@ -23,14 +23,16 @@ class MailActivity(models.Model):
 
     @api.depends("res_model", "res_id")
     def _compute_res_partner_id(self):
-        for obj in self:
-            res_model = obj.res_model
-            res_id = obj.res_id
-            if res_model == "res.partner":
-                obj.partner_id = res_id
-            else:
-                res_model_id = obj.env[res_model].search([("id", "=", res_id)])
-                if "partner_id" in res_model_id._fields and res_model_id.partner_id:
-                    obj.partner_id = res_model_id.partner_id
+        for activity in self:
+            res_model = activity.res_model
+            res_id = activity.res_id
+            activity.partner_id = False
+            if res_model:
+                if res_model == "res.partner":
+                    activity.partner_id = res_id
                 else:
-                    obj.partner_id = None
+                    res_model_id = self.env[res_model].search([("id", "=", res_id)])
```

Comment:
why search and  not browse? In any case, if you want to search, add `limit=1`

Reply:
Seems reasonable. Comment attended.